File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hs = set()
        n = len(nums)
        if n == 1:
            return False
        for i in nums:
            if i in hs:
                return True
            else:
                hs.add(i)
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hs = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;
            if (hs.containsKey(diff)) {
                return new int[] {hs.get(diff), i};
            }
            hs.put(num, i);
        }
        return new int[] {0, 0};
    }
}
